unix/Makefile

```diff
@@ -30,17 +30,23 @@ else
 COPT = -Os #-DNDEBUG
 endif
 
-# On OSX, 'gcc' is a symlink to clang unless a real gcc is installed.
-# The unix port of micropython on OSX must be compiled with clang,
-# while cross-compile ports require gcc, so we test here for OSX and 
-# if necessary override the value of 'CC' set in py/mkenv.mk
-ifeq ($(UNAME_S),Darwin)
-CC = clang
-# Use clang syntax for map file
-LDFLAGS_ARCH = -Wl,-map,$@.map
+ifeq ($(TARGET), freedos)
+ CC = $(DJGPP)/bin/i586-pc-msdosdjgpp-gcc
+ CROSS_COMPILE = $(DJGPP)/i586-pc-msdosdjgpp/bin/
+ CFLAGS_MOD += -DMICROPY_NLR_SETJMP -Dtgamma=gamma -DMICROPY_EMIT_X86=0 -DMICROPY_NO_ALLOCA=1 -DMICROPY_PY_USELECT=0
```

Comment:
Instead of doing this, is it possible to pass CC=... and CROSS_COMPILE=... and CFLAGS_EXTRA=... in the recursive call to make below?